main/src/ca/uwaterloo/flix/language/phase/jvm/BackendObjType.scala

```diff
@@ -379,38 +379,60 @@ object BackendObjType {
     }
   }
 
-  case class Tag(tpe: BackendType) extends BackendObjType with Generatable {
+  case class Tag(elms: List[BackendType]) extends BackendObjType with Generatable {
```

Comment:
So this already preparing for n-ary tags?

Reply:
Yeah exactly - will be its own PR, but here the list is always just one element.